pallets/tinlake-investor-pool/src/lib.rs

```diff
@@ -740,6 +863,106 @@ pub mod pallet {
 			};
 		}
 
+		pub(crate) fn calculate_collect(
```

Comment:
general comment for this function: this is _a lot_ of code and logic in a row. Do you see a way of maybe extracting some parts into self-sustained that can be composed and make this function more readable and easier to follow for non-experts?